filters/austerity.go

```diff
@@ -1,34 +1,65 @@
 package filters
 
 import (
-	"fmt"
 	"math"
 	"math/rand"
 	"sync"
 
 	"github.com/stripe/unilog/clevels"
+	"github.com/stripe/unilog/json"
 )
 
 var startSystemAusterityLevel sync.Once
 
-func AusterityFilter(line string) string {
+// AusterityFilter applies system-wide austerity levels to a log
+// line. If austerity levels indicate a line should be shedded, the
+// like's contents will replaced with "(shedded)" in text mode and the
+// "shedded"=true attribude in JSON mode.
+//
+// Shedding log lines retains their time stamps.
+type AusterityFilter struct{}
+
+// Setup starts the parser for the system austerity level. It is
+// exported so that tests can call it with testing=true in test setup,
+// which will disable sending the austerity level (and also appease
+// the race detector).
+func (a AusterityFilter) Setup(testing bool) {
 	// Start austerity level loop sender in goroutine just once
 	startSystemAusterityLevel.Do(func() {
-		go clevels.SendSystemAusterityLevel()
+		if !testing {
+			go clevels.SendSystemAusterityLevel()
+		}
 	})
+}
 
-	criticalityLevel := clevels.Criticality(line)
-	austerityLevel := <-clevels.SystemAusterityLevel
-	fmt.Printf("austerity level is %s\n", austerityLevel)
+func (a AusterityFilter) FilterLine(line string) string {
+	a.Setup(false)
+	if shouldShed(clevels.Criticality(line)) {
+		return "(shedded)"
```

Comment:
This may interact weirdly with JSON -- I don't expect we want to mix JSON and non-JSON in the same file if possible.

Separately, I think I chatted briefly with Aditya about the possibility of including a sampling percentage in each line emitted? I think this would be more useful than a bunch of extra lines saying nothing, and 9 bytes (+ timestamp) for 90% of shedded lines vs something like 5 for 10% of unshed lines would be less data. (e.g. something like `[10%]` in or after the Unilog timestamp prefix)